marklogic-client-api/src/test/java/com/marklogic/client/test/rows/RowManagerValidateDocTest.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2022 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.client.test.rows;
+
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.marklogic.client.datamovement.DataMovementManager;
+import com.marklogic.client.datamovement.WriteBatcher;
+import com.marklogic.client.document.DocumentManager;
+import com.marklogic.client.document.XMLDocumentManager;
+import com.marklogic.client.expression.PlanBuilder;
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.JacksonHandle;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.io.marker.AbstractWriteHandle;
+import com.marklogic.client.row.RowRecord;
+import com.marklogic.client.test.Common;
+import com.marklogic.client.type.PlanParamExpr;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.util.StringUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.marklogic.client.io.Format.JSON;
+import static com.marklogic.client.io.Format.XML;
+import static org.junit.Assert.*;
+
+public class RowManagerValidateDocTest extends AbstractOpticUpdateTest {
+
+    private Set<String> expectedUris = new HashSet<>();
+    private DataMovementManager dataMovementManager;
+
+    @Before
+    public void moreSetup(){
+        dataMovementManager = Common.client.newDataMovementManager();
+    }
+// TODO : test with rest-writer user when https://bugtrack.marklogic.com/57978 is fixed
+    @Test
+    public void xmlSchema() {
+        expectedUris.clear();
+        WriteBatcher writeBatcher = dataMovementManager.newWriteBatcher();
+        DocumentMetadataHandle meta = newDefaultMetadata();
+        dataMovementManager.startJob(writeBatcher);
+        final int uriCountToWrite = 10;
+        for (int i = 0; i < uriCountToWrite; i++) {
+            String uri = "/acme/" + i + ".xml";
+            writeBatcher.addAs(uri, meta, new StringHandle("<Doc><key>" + i + "</key><Value>value" + i + "</Value></Doc>").withFormat(XML));
+            expectedUris.add(uri);
+        }
+        writeBatcher.flushAndWait();
+        dataMovementManager.stopJob(writeBatcher);
```

Comment:
This test would be simpler and more realistic if it instead wrote only 2 documents, one that is valid one that is invalid. That is sufficient for verifying that the schema is being used correctly to validate each document before it is written. And it avoids the need to use DMSDK to insert N documents first.

Reply:
Also, I think this will be simpler if you just create small thesaurus documents, as shown in the docs at https://docs.marklogic.com/xdmp.validate . That way, you don't need to define an XML schema. And make the docs as small as possible - you just need 1 valid one and 1 invalid one.

Reply:
i tried to send an invalid xml but the transaction does not go through

Reply:
Check with @josvanroosmalen , he's rewriting this processor right now.